chore(ci): switch to JReleaser for migration from Sonatype OSSRH to Central Portal

yuichi0301 opened the pull request:
As part of our migration from the legacy Sonatype OSSRH endpoint to the new Central Portal, this change updates our release process to use JReleaser.  
- Update gradle to 8.7 to use JReleaser
- ✨ Integrate the JReleaser Gradle plugin in `build.gradle`  
  - Enable signing in COMMAND mode  
- 🔧 Update GitHub Actions workflow
  - Run `./gradlew jreleaserFullRelease` instead of manual publish steps 
- 📝 Simplify Maven Central deployment by delegating to JReleaser’s built-in staging/release logic

yuichi0301 committed:
chore(ci): switch to JReleaser for migration from Sonatype OSSRH to Central Portal

yousuketto commented:
@yuichi0301 Thanks for creating this pull request.  I have two comments.

First, is `JRELEASER_PROJECT_VERSION` really necessary for us? It looks like `github.ref_name` is always `master`. Also, I don't think we need to override the version number since it's specified in `build.gradle`.

Second, I noticed the logic for switching publishing URL based on whether the version conttains `-SNAPSHOT` has been removed. I thought this was still necessary. Could you clarify why it was changed?

Commit:
chore: remove JRELEASER_PROJECT_VERSION override since it's handled automatically

yuichi0301 commented:
@yousuketto 
JRELEASER_PROJECT_VERSION certainly doesn't seem necessary, so delete it.
The behavior of switching the publication URL based on the version is controlled by the following section in jreleaser.yml.

```yml
deploy:
  maven:
    nexus2:
      maven-central:
        active: ALWAYS
        url: https://oss.sonatype.org/service/local
        snapshotUrl: https://oss.sonatype.org/content/repositories/snapshots
        stagingRepository:
          release: true
        credentials:
          username: env.OSSRH_USERNAME
          password: env.OSSRH_PASSWORD
        gpg:
          passphrase: env.GPG_PASSPHRASE
```

yousuketto commented:
@yuichi0301 Thanks. I didn't see the `jreleaser.yml`. I have two questions about it.

First, I see that both `jreleaser.yml` and `build.gradle` define a version number. Having the version in two separate files seems like it could lead to human error, since we have to remember to update both for a release. Is that correct? If so, I think we should use a single source of truth for the version number.

Second, are the values under `deploy.maven.nexus2` configured for `OSSRH` URL? If so, they should probably be updated for `Central Portal`.

yuichi0301 committed:
chore: migrate JReleaser deploy config from nexus2 to mavenCentral

- Remove deploy.maven.nexus2.maven-central section
- Add deploy.maven.mavenCentral section to use Central Portal API
- Simplify snapshot settings under mavenCentral

yuichi0301 commented:
@yousuketto 

# Version duplication
Having the version hard-coded in both build.gradle and jreleaser.yml is indeed error-prone. To avoid having to update two places, I’ll remove the version: entry from jreleaser.yml and let JReleaser pick up project.version from Gradle (e.g. via gradle.properties) as a single source of truth.

# OSSRH vs Central Portal URLs
The current deploy.maven.nexus2 block does point at the legacy OSSRH endpoints. I’ll switch that over to the mavenCentral deployer so we use the new Central Portal API instead.

Commit:
Configure project for SNAPSHOT deployment to Central Portal

Commit:
Fix to master branch

yuichi0301 commented:
@yousuketto 
I have fixed it so that the snapshot can be released. Please confirm.
